test/Riok.Mapperly.Tests/Mapping/EnumExplicitMapTest.cs

```diff
@@ -0,0 +1,291 @@
+using Riok.Mapperly.Diagnostics;
+
+namespace Riok.Mapperly.Tests.Mapping;
+
+public class EnumExplicitMapTest
+{
+    [Fact]
+    public void EnumByNameWithExplicitValue()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapEnumValue(E2.e, E1.E), MapEnum(EnumMappingStrategy.ByName)] public partial E1 ToE1(E2 source);",
+            "public enum E1 {A, B, C, D, E, f, F}",
+            "public enum E2 {A = 100, B, C, d, e, E, f}"
+        );
+        TestHelper
+            .GenerateMapper(source, TestHelperOptions.AllowInfoDiagnostics)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                return source switch
+                {
+                    global::E2.A => global::E1.A,
+                    global::E2.B => global::E1.B,
+                    global::E2.C => global::E1.C,
+                    global::E2.e => global::E1.E,
+                    global::E2.E => global::E1.E,
+                    global::E2.f => global::E1.f,
+                    _ => throw new System.ArgumentOutOfRangeException(nameof(source), source, "The value of enum E2 is not supported"),
+                };
+                """
+            )
+            .HaveDiagnostic(new(DiagnosticDescriptors.SourceEnumValueNotMapped, "Enum member d (103) on E2 not found on target enum E1"))
+            .HaveDiagnostic(new(DiagnosticDescriptors.TargetEnumValueNotMapped, "Enum member D (3) on E1 not found on source enum E2"));
```

Comment:
Should there not be a diagnostic for `"Enum member F (6) on E1 not found on source enum E2"` ?

Reply:
Improved.